Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java

```diff
@@ -65,24 +108,77 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         return view;
     }
 
+    @OptIn(markerClass = ExperimentalBetaApi.class)
     @Override
     public void onResume() {
         super.onResume();
         PayPalPendingRequest.Started pendingRequest = getPendingRequest();
         if (pendingRequest != null) {
-            PayPalPaymentAuthResult paymentAuthResult = payPalLauncher.handleReturnToApp(pendingRequest, requireActivity().getIntent());
-            if (paymentAuthResult instanceof PayPalPaymentAuthResult.Success) {
-                completePayPalFlow((PayPalPaymentAuthResult.Success) paymentAuthResult);
+            PayPalPaymentAuthResult paymentAuthResult = payPalLauncher.handleReturnToApp(
+                    pendingRequest,
+                    requireActivity().getIntent()
+            );
+
+            if (paymentAuthResult != null) {
+                if (paymentAuthResult instanceof PayPalPaymentAuthResult.Success) {
+                    completePayPalFlow((PayPalPaymentAuthResult.Success) paymentAuthResult);
+                } else {
+                    handleError(new Exception("User did not complete payment flow"));
+                }
+            }
+
+            clearPendingRequest();
+        }
+
+        PayPalVaultEditPendingRequest.Started pendingRequestEditFi = getPendingRequestEditFi();
+
+        if (pendingRequestEditFi != null) {
+            PayPalVaultEditAuthResult editAuthResult = payPalLauncher.handleReturnToApp(
+                    pendingRequestEditFi,
+                    requireActivity().getIntent()
+            );
+
+            if (editAuthResult != null) {
+                if (editAuthResult instanceof PayPalVaultEditAuthResult.Success) {
+                    PayPalVaultEditResult result = payPalClient.edit(
+                            (PayPalVaultEditAuthResult.Success) editAuthResult
+                    );
+
+                    handleVaultEditParsing(result);
+
+                } else if (editAuthResult instanceof PayPalVaultEditAuthResult.Failure) {
+                    handleError(((PayPalVaultEditAuthResult.Failure) editAuthResult).getError());
+                } else if (editAuthResult instanceof PayPalVaultEditAuthResult.NoResult) {
+                    handleError(new Exception("User did not complete payment flow"));
+                }
             } else {
                 handleError(new Exception("User did not complete payment flow"));
             }
-            clearPendingRequest();
+
+            clearEditFiPendingRequest();
+        }
+    }
+
+    @OptIn(markerClass = ExperimentalBetaApi.class)
+    private void handleVaultEditParsing(PayPalVaultEditResult result) {
```

Comment:
Let's call this `completeEdit` or something that exposes less internals since this is a sample merchant integration

Reply:
And move the method invocation (`PayPalClient.edit`) into this method